stores/metadata.go

```diff
@@ -1510,6 +1607,9 @@ func deleteObject(tx *gorm.DB, path string) (numDeleted int64, _ error) {
 		return 0, tx.Error
 	}
 	numDeleted = tx.RowsAffected
+	if numDeleted == 0 {
+		return 0, nil // nothing to prune if no object was deleted
```

Comment:
unrelated to this line but it made me think about it: what happens to a buffered slab when the slab it's referencing gets deleted, e.g. what happens when buffered data is deleted, there's no cleanup mechanism right?